### ISSUE TITLE
Some players getting competitive rank from previous seasons

### ISSUE THREAD

**Issue by justinkhart** [justinkhart/OverwatchChumps#11]
@studro reported this issue, need more details (which player).

**Comment by studro** [comment]
Perseus hasn't ranked yet this season. Funnily enough, playoverwatch.com is also showing last season's rank for his battle tag.

**Comment by justinkhart** [comment]
I did a bit of research and it looks like thats what the overwatch website does (the source of all the stats). We can't fix this, think of it more like a feature. :)
https://playoverwatch.com/en-us/career/pc/us/Perseus-1381

**Comment by studro** [comment]
Nah, we definitely can fix it (for most cases at least) with this heuristic:

We take any rank-reading from in the "gap week" between last season and the current season for a given player.

If there doesn't exist a ranking after that date that differs from the initial reading, that means the player is unranked.

Some pseudocode (ignore the incorrect identifiers)

    SELECT TOP 'RANKED' FROM rank WHERE 
    rank <> (
        SELECT rank FROM rank WHERE 
        timestamp = #someTimestampDateInLastGapWeek# AND 
        playerId = #currentPlayerId# ) AND
    timestamp >= #someTimestampDateInLastGapWeek# AND 
    playerId = #currentPlayerId#

To work out `#someTimestampDateInLastGapWeek#`, we can just

    SELECT TOP timestamp FROM rank WHERE timestamp >= #startGapWeek# AND timestamp < #endGapWeek# AND playerId = #currentPlayerId#

**Comment by justinkhart** [comment]
Yeah nice good thinking. OK I'll leave this one open.
